zkevm-circuits/src/witness/block.rs

```diff
@@ -282,6 +284,69 @@ impl Block {
         log::debug!("tx_log num: {}", self.rws.rw_num(RwTableTag::TxLog));
         log::debug!("start num: {}", self.rws.rw_num(RwTableTag::Start));
     }
+
+    // This helper returns bytecodes's whether `code_hash` is belong to first bytecode circuit.
+    // always return true when feature 'dual_bytecode' is disabled.
+    pub(crate) fn is_first_sub_bytecode_circuit(&self, code_hash: &U256) -> bool {
+        // bytecode_map should cover the target 'code_hash',
+        // but for extcodecopy, the external_address can be non existed code hash.
+        // `unwrap` here is not safe.
+        if self.bytecode_map.is_none() {
+            // not config feature 'dual_bytecode' case.
+            true
+        } else {
+            let bytecode_map = self
+                .bytecode_map
+                .as_ref()
+                .expect("bytecode_map is not none when enable 'dual_bytecode' feature");
+            *bytecode_map.get(code_hash).unwrap_or(&true)
+        }
+    }
+
+    // Get two sets of bytecodes for two bytecode sub circuits when enable feature 'dual_bytecode'.
+    pub(crate) fn get_bytecodes_for_dual_sub_circuits(&self) -> (Vec<&Bytecode>, Vec<&Bytecode>) {
```

Comment:
at the beginning, they are gated by feature, after changed to check feature by  `bytecode_map.is_some()`, don't need explicit gating now. also reduce feature` cfg!(feature= ` lines.

Reply:
In general, if a method is never used if the feature is not enabled, then the method should be feature gated.

Reply:
`bytecode_map` is `Some` if the feature is enabled is the behaviour today. But in the future that behaviour could be slightly modified, and in that case, we don't wish to leak regression issues if all places are not updated appropriately. So we should prefer consistency in how we check for a condition. In this case, I felt the best is to use the `cfg(feature = "dual-bytecode")`, instead of the "value" of the bytecode map (whether `Some` or `None`)

Reply:
if so, this will go back old version in https://github.com/scroll-tech/zkevm-circuits/pull/1368/commits/d61401ad7d46276c71646f30c622713a5c18046c#diff-e71dae4a71b5661aaa2385201c4554b7d56790191c6e5d0a54ebdcac4c63dd92 , thus will import many feature config checking codes, we think not good so remove them and use this way now.